log.go

```diff
@@ -95,42 +95,53 @@ func (hook *logrusFileHook) Levels() []logrus.Level {
 	}
 }
 
-// StartWritingStats writes fm.Stats every minute to Config.StatsFile
+// startWritingStats writes fm.Stats every minute to Config.StatsFile
 // This method should only be called once
-func (fm *Frontman) StartWritingStats() {
-	var stats stats.FrontmanStats
+func (fm *Frontman) startWritingStats() {
+
+	// Only start writing out stats if there is a StatsFile configued
+	if fm.Config.StatsFile == "" {
+		return
+	}
+
+	fm.stats.StartedAt = time.Now()
+	logrus.Debugf("Start writing stats file: %s", fm.Config.StatsFile)
+
 	var buff bytes.Buffer
 	var err error
 
 	// Make the output indented
 	encoder := json.NewEncoder(&buff)
 	encoder.SetIndent("", "    ")
 
-	// Only start writing out stats if there is a StatsFile configued
-	if fm.Config.StatsFile != "" {
-		go func() {
-			for {
-				buff.Reset()
-				time.Sleep(time.Minute * 1)
-				stats.Uptime = uint64(time.Since(stats.StartedAt).Seconds())
-				// Get snapshot from current stats
-				stats = *fm.Stats
-				err = encoder.Encode(stats)
-				if err != nil {
-					logrus.Errorf("Could not encode stats file: %s", err)
-					continue
-				}
-
-				err = ioutil.WriteFile(fm.Config.StatsFile, buff.Bytes(), 0755)
-				if err != nil {
-					logrus.Errorf("Could not write stats file: %s", err)
-					return
-				}
-			}
-		}()
+	for {
+		buff.Reset()
+		time.Sleep(time.Minute * 1)
```

Comment:
usually such things are configurable as you cannot assume if the sleep interval is too big or too less in a concrete use case

Reply:
This is just recycled code from the refactor.
Additionally this code just has a single use case, which is to write stats to disk on a regular interval. Apparently it was set to 1 minute.

Reply:
the ask is to make it configurable.

Reply:
Yea, I understand that. I'm just saying the code is not new with this PR and has not changed either, so making it configurable is outside the scope of task DEV-1757

https://github.com/cloudradar-monitoring/frontman/pull/173/files/9b6fc3ba5a71a2376180f85c564ef7131adf778e#diff-973afd7a3fa6644d1e62a8846c6ee7dde87077ff5c0ee193624f98154eb77687L112-L114